components/humantask/org.wso2.carbon.humantask/src/main/java/org/wso2/carbon/humantask/core/api/mgt/HTQueryBuildHelperImpl.java

```diff
@@ -0,0 +1,628 @@
+        TPredefinedStatus.Enum.table.forInt(1);
+        org.apache.xmlbeans.StringEnumAbstractBase.Table table = TPredefinedStatus.Enum.table;
+        statuses = new TPredefinedStatus.Enum[statusesSize];
+        String[] statusCounts = new String[statusesSize];
+        for (int i = 0; i < statusesSize; i++) {
+            statuses[i] = TPredefinedStatus.Enum.forInt(i + 1);
+        }
+        Boolean isTaskName = false;
+        Pattern p = Pattern.compile("-[0-9]+$");
+        Matcher m = p.matcher(taskName);
+        isTaskName = m.find();
+        if (isTaskName) {
+            for (int i = 0; i < statuses.length; i++) {
+                statusCount1 = getTaskCountsForStateAndTaskName(taskName, statuses[i]);
+                statusCounts[i] = " " + statuses[i].toString() + " " + statusCount1;
+            }
+        } else {
+            for (int i = 0; i < statuses.length; i++) {
+                statusCount1 = getTaskCountsForStateAndTaskDefName(taskName, statuses[i]);
+                statusCounts[i] = " " + statuses[i].toString() + " " + statusCount1;
+            }
+        }
+        return statusCounts;
+    }
+
+    /**
+     * @param taskname
+     * @return
+     * @throws IllegalAccessFault
+     * @throws IllegalArgumentFault
+     * @throws IllegalStateFault
+     * @throws IllegalOperationFault
+     */
+    public int getTaskInstanceCountForTaskName(final String taskname)
+            throws IllegalAccessFault, IllegalArgumentFault, IllegalStateFault,
+                   IllegalOperationFault {
+        final int[] taskCountTemp = {0};
+        final List[] tasklist = new List[1];
+
+        try {
+
+            HumanTaskServiceComponent.getHumanTaskServer().getTaskEngine().getScheduler().
+                    execTransaction(new Callable<Object>() {
+                        public Object call() throws Exception {
+
+                            HumanTaskServerHolder.getInstance().getHtServer().getTaskEngine().getDaoConnectionFactory().getConnection();
+                            HumanTaskServer humanTaskServer = HumanTaskServiceComponent.getHumanTaskServer();
+                            HumanTaskEngine humanTaskEngine = humanTaskServer.getTaskEngine();
+                            HumanTaskDAOConnectionFactory humanTaskDAOConnectionFactory = humanTaskEngine.getDaoConnectionFactory();
+                            humanTaskServer.getServerConfig();
+                            HTQueryBuilder htQueryBuilder = new HTQueryBuilder();
+                            EntityManager em = humanTaskDAOConnectionFactory.getConnection().getEntityManager();
+                            Query query = htQueryBuilder.buildQueryToFindTaskInstances(taskname, em);
+                            tasklist[0] = query.getResultList();
+                            taskCountTemp[0] = tasklist[0].size();
+                            return null;
+                        }
+                    });
+        } catch (Exception ex) {
```

Comment:
Fixed in the latest commit. Moved all exception catches to the top most level of the code.